ksqldb-engine/src/test/java/io/confluent/ksql/engine/rewrite/AstSanitizerTest.java

```diff
@@ -196,6 +203,35 @@ public void shouldAddQualifierForJoinColumnReferenceFromRight() {
     ))));
   }
 
+  @Test
+  public void shouldSanitizeLambdaArguments() {
+    // Given:
+    final Statement stmt = givenQuery(
+        "SELECT TRANSFORM_ARRAY(X => X + 5) FROM TEST2;");
```

Comment:
Is this a valid statement in practice? I.e. without a column reference of the entity `TEST`, how should we know which column to apply this lambda expression?

Reply:
This particular one  wouldn't be valid.

We don't have to have a column reference though to apply the invocation function, we could have `Select Transform_Array(ARRAY[1, 2], x => x + 5) FROM TEST2 emit changes;`

Reply:
What does `ARRAY[1, 2]` mean here?

Reply:
https://docs.ksqldb.io/en/latest/developer-guide/syntax-reference/#array

It's how we create an inline array in the language. In this case we're not using any of the rows from the source and instead just using a constant array for use in the function

Reply:
I see. So in this case it is irrelevant to `TEST2`'s records, and would always return a single record as {key: null, value: [6,7]} right?

Reply:
yup